tfhe/src/core_crypto/gpu/slice.rs

```diff
@@ -158,12 +158,8 @@ where
             None
         } else {
             // Shift ptr
-            let shifted_ptr: *mut c_void = unsafe {
-                self.ptrs[gpu_index as usize]
-                    .cast::<u8>()
-                    .add(start * std::mem::size_of::<T>())
-                    .cast()
-            };
+            let shifted_ptr: *mut c_void =
+                unsafe { self.ptrs[gpu_index as usize].byte_add(start * std::mem::size_of::<T>()) };
```

Comment:
wouldn't https://doc.rust-lang.org/std/primitive.pointer.html#method.add-1 be better to avoid having to manually multiplying by some constant ?

Reply:
ah damn missed that